tuf/src/error.rs

```diff
@@ -45,11 +47,35 @@ pub enum Error {
         uri: String,
     },
 
+    /// An IO error occurred.
+    #[error("IO: {0}")]
+    Io(#[from] io::Error),
+
+    /// A json serialization error occurred.
+    #[error("json: {0}")]
+    Json(#[from] serde_json::error::Error),
+
     /// Errors that can occur parsing HTTP streams.
     #[cfg(any(feature = "hyper_013", feature = "hyper_014"))]
     #[error("hyper: {0}")]
     Hyper(#[from] hyper::Error),
 
+    /// A tempfile persist error occurred.
+    #[error("tempfile: {0}")]
+    TempfilePersist(#[from] tempfile::PersistError),
+
+    /// A tempfile path persist error occurred.
+    #[error("tempfile: {0}")]
+    TempfilePathPersist(#[from] tempfile::PathPersistError),
+
+    /// A derp error occurred.
+    #[error("derp: {0}")]
```

Comment:
"derp" is a cute name, but maybe we should indicate what that means in the error string. "der parse: {0}"?